### ISSUE TITLE
Option to automatically pick recommended when something is not set instead of prompt

### ISSUE THREAD

**Issue by MahouShoujoMivutilde** [Frogging-Family/linux-tkg#242]
I think it will be useful for automatic periodic builds - when something was changed / new option is added, etc.

it is also nice to not have to specify everything explicitly, and only change things that are relevant to you.

**Comment by AdelKS** [comment]
There is such a thing, it's the variable `_OPTIPROFILE`. Although Linux-tkg is more about choice, as each system is unique and what is good for one system is not for another. Testing stuff out is the only way to actually know what's good for your system, there's no running away from it.

**Comment by MahouShoujoMivutilde** [comment]
Of course! I didn't meant it as if i wanted everything configured for me, no customization involved.

No. What i meant is that if i didn't explicitly set something as false (well, maybe i didn't see it yet / maybe i don't want it) - it would be nice to have an option to just assume i want default for it.

Like, everything i wanted, i turned on / selected, whatever will be added later - i will check, but don't ask me during build about it, just assume i want something sensible (as if i just pressed enter for default value when prompted).

That way it would be easier to setup automatic builds on a personal server, since right now it sometimes breaks because it hangs waiting for answer.

> `_OPTIPROFILE`

I know about it, but it still does not prevent prompts for other options, e.g.

```
# MuQSS and PDS only - SMT (Hyperthreading) aware nice priority and policy support (SMT_NICE) - Kernel default is "true" - You can disable this on non-SMT/HT CPUs for lower overhead
_smt_nice=""
```

...will still prompt me for choice even when some `_OPTIPROFILE` is selected (unless this `_OPTIPROFILE` already sets `_smt_nice`, which is not the point, since to write such file i should already know about `_smt_nice`, and in such case - why not just set it in `customization.cfg` in first place?)

I mean, it says

> `Kernel default is "true"`

so what i want, is an option that would just pick that default without prompt.

**Comment by AdelKS** [comment]
Okay I understand better what you want. It may be interesting to have a `_no_prompt` config option. Let's hear what others have to say.

One possible workflow that you can already use is to actually edit entirely `customization.cfg` and update the repository with `git stash && git pull && git stash pop` so your customization file is kept. And if the command fails, which it shouldn't happen frequently since config options usually do not get touched ; or you get prompted after that, then just edit it again.

About `_OPTIPROFILE` prompting, I do not know if it's really meant to avoid prompts or not. Let's wait for some more feedback.

**Comment by MahouShoujoMivutilde** [comment]
Yeah, that's very similar to what i already use, i just had to re-setup it in other place, and this came up again. So i thought i'd share. 

> About `_OPTIPROFILE` prompting

I don't think it is, to me it reads more like - okay, you have this kind of system, so you probably want that, that, and that, and we'll ask you about other things later.

**Comment by AdelKS** [comment]
Now I think you can at least pipe empty lines to the install script, something like `yes "" | makepkg -si` or `yes "" | ./install.sh install`. You need to setup the distro variable and CPU scheduler as they do not have a default entry.

**Comment by MahouShoujoMivutilde** [comment]
Sorry for the late reply. 

This works, but only for makepkg directly, when building in chroot (systemd-nspawn via aurutils) it doesn't seem to reach the prompts. 

So i still would prefer having option to auto-select defaults.

**Comment by AdelKS** [comment]
`yes "" | ./install.sh install` definitely works now, well you still need to set some important variables from the customization in the env. That's what we do in the ci.
